### DISCUSSION TITLE
How do you add parameters to the rest endpoint ?

### DISCUSSION THREAD

**Discussion by xclaim494** [Azure/data-api-builder#2320] [Q&A] [answered]
I have a View called 'Quotes_V'

```
columns are :
          "underlying",
		  "option_type",
		  "strike",
		  "expiration_date",
		  "bid",
		  "ask"
```

i need to build a rest endpoint on the quotes View with parameters for "underlying", "option_type", "strike", "expiration_date" and the payload returned needs to be bid, ask (+ others)

I am absolutely new to this so please excuse my ignorance : 

I got as far as building a simple config file with a key-fields like below :

```
"entities": {
    "Quote": {
      "source": {
        "object": "dbo.Quote_V",
        "type": "view",
        "key-fields": [
          "underlying",
		  "option_type",
		  "strike",
		  "expiration_date"
        ]
```

when i run the rest URL I use : 
http://localhost:5000/api/Quote/underlying/aapl/option_type/call/strike/100/expiration_date/2024-07-12

and the payload that is retuned is correct :

> {
>   "value": [
>     {
>       "bid": 127.2,
>       "ask": 128.64999
>     }
>   ]
> }

my question is : does the rest url look right ? I thought the parameters to query the fields needed to be structured differently ?

any guidance is much appreciated

example in curl I am used to something like this ?
`curl -X GET "http://localhost:5000/api/Quote/?underlying=aapl&option_type=call&strike=100&expiration_date=2024-07-12"`

**Comment by seantleonard** [comment]
Hi @xclaim494, the 'key-fields' can be thought of as "primary keys." Wouldn't the primary key in this case be `underlying`? then your REST url can use the filter query parameter to filter results:
```https
GET  https://localhost:5001/api/Quote/underlying/aapl?$filter=option_type eq 'call' and strike eq 100 and expiration_date eq '2024-7-12'

```

https://learn.microsoft.com/en-us/azure/data-api-builder/rest#filter

**Comment by sezal98** [comment, marked as answer]
Hi @xclaim494 , As mentioned above we add primary keys in the REST URL like 
<img width="460" alt="image" src="https://github.com/user-attachments/assets/4ba8ed9b-73c1-40aa-aad5-1adc316df164" />
and the fields which need to be added go as request body like 
<img width="206" alt="image" src="https://github.com/user-attachments/assets/59eed741-aae9-4010-8d9e-0acc356ff0d5" />

Try running the DAB engine in Development mode and look for /swagger document. This will give you the idea on what URLs would be executed and with what parameters. 

For example: Please refer to this https://learn.microsoft.com/en-us/azure/data-api-builder/rest#put 
Here the primary key for Table Book is column ID which is 2001 and the other fields like title and pages go in the body when executing the POST or PUT command.
